### PR TITLE
Plane: add param for throttle control in transition to VTOL

### PR DESCRIPTION
**Pull request by Hwurzburg**
Allows either hover throttle or user param for throttle level in tailsitter back transistion...can be used to make a flat transition (when the z controller gets fixed.... with master it can reduce  zoom a lot)

### CODE REVIEW

File: ArduPlane/tailsitter.cpp [hunk 1]
```diff
@@ -198,13 +206,21 @@ void Tailsitter::output(void)
...
               integrator to the same throttle level
               convert the hover throttle to the same output that would result if used via AP_Motors
               apply expo, battery scaling and SPIN min/max. 
             */
-            throttle = motors->thrust_to_actuator(motors->get_throttle_hover()) * 100;
-            throttle = MAX(throttle,plane.aparm.throttle_cruise.get());
+            if (!is_negative(transition_throttle_vtol)) { 
+                // Q_TAILSIT_THR_VT is positive use it until transition is complete
+                throttle = MIN(transition_throttle_vtol,100.0);
+                throttle = motors->thrust_to_actuator(throttle);
```
Comment (IamPete1) [inline ArduPlane/tailsitter.cpp:217]: I don't think we need this. Firstly it is expecting a 0-1 value and secondly this applies throttle expo, I think that just makes it less clear to the user that it is doing the throttle they asked for. Not that fussed if you think hving the expo is better in that case it should be:

`throttle = motors->thrust_to_actuator(throttle * 0.01) * 100;`

File: ArduPlane/tailsitter.cpp [hunk 2]
```diff
@@ -413,7 +429,6 @@ bool Tailsitter::transition_vtol_complete(void) const
         return true;
     }
     // still waiting
-    quadplane.attitude_control->reset_rate_controller_I_terms();
```
Comment (IamPete1) [inline ArduPlane/tailsitter.cpp:416]: this change is not related to this PR.

File: ArduPlane/tailsitter.cpp (hunk 1 above)
Comment (Hwurzburg) [reply ArduPlane/tailsitter.cpp:217]: if its good for hover throttle, why not a user value? battery comp,etc...

File: ArduPlane/tailsitter.cpp (hunk 2 above)
Comment (Hwurzburg) [reply ArduPlane/tailsitter.cpp:416]: oops...will fix..branched off wrong place apparently

File: ArduPlane/tailsitter.cpp [hunk 3]
```diff
@@ -141,6 +141,14 @@ const AP_Param::GroupInfo Tailsitter::var_info[] = {
     // @Range: 10 500
     AP_GROUPINFO("RAT_VT", 17, Tailsitter, transition_rate_vtol, 50),
 
+    // @Param: THR_VT
+    // @DisplayName: Tailsitter forward flight to VTOL transition throttle
+    // @Description: Throttle used during FW->VTOL transition, -1 uses hover throttle
+    // @Units: percent
```
Comment (IamPete1) [inline ArduPlane/tailsitter.cpp:147]:
```suggestion
    // @Units: %
```
this should get it past CI

File: ArduPlane/tailsitter.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/tailsitter.cpp]: Fair point about battery comp, maybe it is better to have the corrected value.

File: ArduPlane/tailsitter.cpp (hunk 3 above)
Comment (Hwurzburg) [reply ArduPlane/tailsitter.cpp]: thanks...done....I removed the compensation becuase even if the thr was set for 15% it pushed it up to over 25% on a full battery....the non battery comp stuff must be messing with it...is there a batt comp only function to alter throttle without the expo and other curve adjustments?

Comment (IamPete1) [reply ArduPlane/tailsitter.cpp]: I think we actually have a bug here. For the assist case we are applying the linearization to throttle twice.

**Comment by IamPete1** [comment]
https://github.com/ArduPilot/ardupilot/pull/18428 should fix the issue when using expo correction. We now only apply it once.
